Updating output checking processes (#1246)
opensafely/documentation: docs/releasing-files.md

### Rounding rates

> A rate consists of a numerator and denominator, which are generally both counts. **In OpenSAFELY, any rate calculated from counts below the redaction threshold (5 or less) should also be a redacted.** In addition, we recommend rounding because redaction alone is vulnerable to differencing. When future calculations rely on rates not being mapped to a non-numerical like `[REDACTED]` and/or a distinction between a rate of zero and a non-zero rate is desirable, we recommend rounding the numerator and denominator to 'midpoint 6'. In short, rounding to 'midpoint 6' allows differentiating between zero and non-zero rates, by not breaking our suppression rules and without introducing bias.

Reviewer: ```suggestion
A rate consists of a numerator and denominator, which are generally both counts. **In OpenSAFELY, any rate calculated from counts below the redaction threshold (5 or less) should also be redacted.** In addition, we recommend rounding because redaction alone is vulnerable to differencing. When future calculations rely on rates not being mapped to a non-numerical like `[REDACTED]` and/or a distinction between a rate of zero and a non-zero rate is desirable, we recommend rounding the numerator and denominator to 'midpoint 6'. In short, rounding to 'midpoint 6' allows differentiating between zero and non-zero rates, by not breaking our suppression rules and without introducing bias.
```

Each section in the review request form should normally describe a single file, but where necessary for similar files, these can be grouped together and wildcards can be used for the file path (e.g. `release/hospitalisation_rate_by_*.csv`). **If you use a wildcard, please indicate how many files this captures**.

### Release of intermediate data

In general, releases should be for final results from your project (see the note above). However, on some occassions it is appropriate to release intermediate data. Below are some suggestions for when this is appropriate:

* You think you may need to make minor edits to final outputs such as changing figure labels. Release of the intermediate data allows you to make these changes locally.

> * A large number of outputs are produced from a single intemediate output. Release of the intermediate data underlying the figures (which needs to be checked whether it is released or not) avoids the need to check the downstream outputs.

Reviewer: ```suggestion
* A large number of outputs are produced from a single intermediate output. Release of the intermediate data underlying the figures (which needs to be checked whether it is released or not) avoids the need to check the downstream outputs.
```

> *  The intemediate data doesn't contain person-level data, but is used for running a model that would produce multiple outputs.

Reviewer: ```suggestion
*  The intermediate data doesn't contain person-level data, but is used for running a model that would produce multiple outputs.
```

If requesting release of intermediate data there are a few considerations:

> *  We recommend that you continue to develop downstream analysis actions within the OpenSAFELY pipeline, even if they are not inteneded to be run on the server against. This helps maintain reproducibility.

Reviewer: ```suggestion
*  We recommend that you continue to develop downstream analysis actions within the OpenSAFELY pipeline, even if they are not intended to be run on the server. This helps maintain reproducibility.
```

    - transcribing (e.g., to paper or email)
    - using screen sharing software or any recording device/software

### Running further analyses on released outputs

If you have had [intermediate data released](#release-of-intermediate-data) and you wish to run further analyses on them, such as reformatting figures, there are a few things to consider.

1. You should include the code for these steps in your GitHub repo.

> 2. You **should not** commit any of the released outputs to your GitHub repo. Make sure to include them in the `.gitignore` file.

Reviewer: what about the final processed charts/tables created as a result, should they be excluded from the repo as well?

Author: I think they have to be, for now, given our approaches to making repos public and the template readme content. Ultimately, we'll need to be able to upload results to the jobs site
